pkg/policy/api/fqdn.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+type FQDNSelector struct {
+	FQDN []string `json:"fqdn",omitempty`
```

Comment:
What's the rational for the 2nd level slice nesting? The following would make sense to me as well:

```
  toFQDNs:
   - matchName: foo.com
   - matchName: bar.com
```
 (Based on the `matchLabels` of the LabelSelector`

Reply:
It's mostly to leave us room for:
- `except` clauses (for when we support wildcards)
- TTL and caching options that we might add

which would have to live on the same level as `fqdn`/`matchName` but where you might want multiple "names" with the same settings. It's definitely funny looking today, however.
I like `matchName` better so I'll switch to that anyway. I can simplify this, for now, by using a field named `matchName` on the ToFQDNs (`FQDNSelector`) object. This would then look like your example. Later, when we add the other things, we would have to add a new `matchNames` field to allow specifying multiple names that have 1 exception/ttl/cache option. Would that work better?

Reply:
That sounds good.